### PR TITLE
Prepare daily CI for torch 2.0.0

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

~~⚠️⚠️ Don't merge before I run it (again) ⚠️⚠️~~

This PR changes docker files / workflow files to use the upcoming torch `2.0.0`.

### CODE REVIEW

File: .github/workflows/build-docker-images.yml [hunk 1]
```diff
@@ -22,6 +22,16 @@ jobs:
...
+      - name: Cleanup disk
+        run: |
+          sudo ls -l /usr/local/lib/
+          sudo ls -l /usr/share/
+          sudo du -sh /usr/local/lib/
+          sudo du -sh /usr/share/
+          sudo rm -rf /usr/local/lib/android
+          sudo rm -rf /usr/share/dotnet
+          sudo du -sh /usr/local/lib/
+          sudo du -sh /usr/share/
```
Comment (ydshieh) [inline .github/workflows/build-docker-images.yml:34]: When I tried to `install 2.0.0`, the runner ran out of disk. This was the way that fixed the problem.

File: docker/transformers-all-latest-gpu/Dockerfile [hunk 2]
```diff
@@ -24,13 +24,15 @@ ARG REF=main
...
-RUN [ "$PYTORCH" != "pre" ] && python3 -m pip install --no-cache-dir -U $VERSION torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/$CUDA || python3 -m pip install --no-cache-dir -U --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/$CUDA
+## TODO: Handle these in a python utility script
+#RUN [ ${#PYTORCH} -gt 0 -a "$PYTORCH" != "pre" ] && VERSION='torch=='$PYTORCH'.*' ||  VERSION='torch'; echo "export VERSION='$VERSION'" >> ~/.profile
+#RUN echo torch=$VERSION
+## `torchvision` and `torchaudio` should be installed along with `torch`, especially for nightly build.
+## Currently, let's just use their latest releases (when `torch` is installed with a release version)
+## TODO: We might need to specify proper versions that work with a specific torch version (especially for past CI).
+#RUN [ "$PYTORCH" != "pre" ] && python3 -m pip install --no-cache-dir -U $VERSION torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/$CUDA || python3 -m pip install --no-cache-dir -U --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/$CUDA
+
```
Comment (ydshieh) [inline docker/transformers-all-latest-gpu/Dockerfile:34]: When the official `torch 2.0.0` (link) is out, we can switch back to the above block.

**Review by sgugger — APPROVED** [review]
Thanks!

File: docker/transformers-all-latest-gpu/Dockerfile [hunk 3]
```diff
@@ -24,13 +24,15 @@ ARG REF=main
...
+RUN python3 -m pip install --no-cache-dir -U torch torchvision torchaudio --index-url https://download.pytorch.org/whl/test/cu117
```
Comment (ydshieh) [inline docker/transformers-all-latest-gpu/Dockerfile:35]: specific to `torch 2.0.0`. Note the `../test/..` part in the link

File: docker/transformers-all-latest-gpu/Dockerfile [hunk 4]
```diff
@@ -54,8 +56,8 @@ RUN python3 -m pip install --no-cache-dir bitsandbytes
 # For video model testing
 RUN python3 -m pip install --no-cache-dir decord av==9.2.0
 
-# For `dinat` model
-RUN python3 -m pip install --no-cache-dir natten -f https://shi-labs.com/natten/wheels/$CUDA/
+## For `dinat` model
+#RUN python3 -m pip install --no-cache-dir natten -f https://shi-labs.com/natten/wheels/$CUDA/
```
Comment (ydshieh) [inline docker/transformers-all-latest-gpu/Dockerfile:60]: `natten` is not ready with torch 2 yet

File: docker/transformers-all-latest-gpu/Dockerfile (hunk 3 above)
Comment (ydshieh) [inline docker/transformers-all-latest-gpu/Dockerfile:35]: @sgugger Just to let you know I am using `https://download.pytorch.org/whl/test/cu117` (with the `test`) here.
We can change it to the correct link once the official release comes out.
I think it's fine to merge like this.

Comment (sgugger) [reply docker/transformers-all-latest-gpu/Dockerfile:35]: This is  the recommended way from the PyTorch team.

**Pull request merged**